generators/python-bots/index.js

```diff
@@ -1,106 +1,104 @@
 
-            if (answers.python_bot_tpl === 'Request/Reply') {
-                this.fs.copyTpl(
-                    this.templatePath('python/bots/request-reply/requirements.txt'),
-                    this.destinationPath('requirements.txt'),
-                    answers
-                );
-                this.fs.copy(
-                    this.templatePath('python/bots/request-reply/src'),
-                    this.destinationPath('python')
-                );
-                this.fs.copy(
-                  this.templatePath('python/bots/request-reply/listeners'),
-                  this.destinationPath('python/listeners')
-                )
-                this.fs.copyTpl(
-                    this.templatePath('python/bots/request-reply/config.json'),
-                    this.destinationPath('resources/config.json'),
-                    answers
-                );
-                this.fs.copy(
-                    this.templatePath('python/.env'),
-                    this.destinationPath('.env')
-                );
-	    } else if (answers.python_bot_tpl === 'Elements Form') {
-                this.fs.copyTpl(
-                    this.templatePath('python/bots/elements_form/requirements.txt'),
-                    this.destinationPath('requirements.txt'),
-                    answers
-                );
-                this.fs.copy(
-                    this.templatePath('python/bots/elements_form/src'),
-                    this.destinationPath('python')
-                );
-                this.fs.copyTpl(
-                    this.templatePath('python/bots/elements_form/config.json'),
-                    this.destinationPath('resources/config.json'),
-                    answers
-                );
-                this.fs.copy(
-                    this.templatePath('python/.env'),
-                    this.destinationPath('.env')
-                );
-            }
-            /* Install certificate */
-            CertificateCreator.create(this.options.initPrompts.encryption, answers.botusername, answers.botemail);
+            this._logAnswers();
+
+            this._copyTemplateFiles();
 
-            let log_text_completion = ('* BOT generated successfully!!').bold;
-            console.log(log_text_completion.bgGreen.white);
+            /* Install certificate */
+            CertificateCreator.create(this.answers.encryption, this.answers.botusername, this.answers.botemail);
+        }).then(() => {
+            let logTextCompletion = ('* BOT generated successfully!!').bold;
+            console.log(logTextCompletion.bgGreen.white);
         });
     }
+
+    // Private methods are prepended with '_' to prevent yeoman from executing them
+    // yeoman's behavior is to execute every method in a generator which does not begin with _
+    _updateWithBotTemplateAnswer(answers) {
+        Object.assign(this.answers, answers);
+        this.baseTemplatePath = (this.answers.pythonBotTemplate === REQUEST_REPLY) ?
```

Comment:
I prefer using switch case than a single condition here in case we can have more than 2 template in future. (For example, we have 4 templates for java-bot)

Reply:
I updated the PR according to your proposal